renderer/src/geometry.rs

```diff
@@ -170,7 +170,7 @@ impl Frame {
 
     /// Applies a scaling to the current transform of the [`Frame`].
     #[inline]
-    pub fn scale(&mut self, scale: f32) {
+    pub fn scale(&mut self, scale: Vector) {
```

Comment:
Uniform scaling seems to be the most common use case. Instead of making the simple use case more complicated, I think a new `scale_nonuniform` method for advanced usage would be better.

We can also make `scale` take an `impl Into<f32>` and `scale_nonuniform` an `Into<Vector>`. This way, we'll be able to call both nicely:

```rust
frame.scale(2);
frame.scale_nonuniform([2.0, 3.0]);
```

Reply:
I agree with adding `scale_nonuniform` rather than changing `scale`.

I'm not sure this PR is the best place for changing the parameters to `impl Into<_>` though since that seems like more of a global iced API issue rather than something to add ad-hoc in a vaguely related PR.